src/octoprint/static/js/app/viewmodels/gcode.js

```diff
@@ -419,6 +419,9 @@ $(function() {
                     self.layerSlider.slider("disable");
                     self.layerSlider.slider("setMax", 1);
                     self.layerSlider.slider("setValue", 0);
+                    $('#btn_layer_up').prop('disabled', true);
```

Comment:
Hm... While there are still some places within OctoPrint where there are such direct jquery references, in general it's less repetitive and cleaner to fetch those elements once and store them on the view model (e.g. like with `self.layerSlider`).

In that specific case here you manipulate the `disabled` property only (plus set a click handler further down below, see the comment to that), and I wonder if that wouldn't work better through a separate property on the viewmodel (e.g. `self.layerSelectionEnabled = ko.observable(false)`) and then setting both buttons' [enabled binding](http://knockoutjs.com/documentation/enable-binding.html) to that within the HTML markup (e.g. `data-bind="enabled: layerSelectionEnabled"`).

Would have the advantage to better separate view from view model (and in fact, the only reason the `self.layerSlider` element gets manipulated directly is because I didn't jump through the hoops of implementing a custom knockout binding for it, would also have been the cleaner approach actually and might be something to do in the future).